jujugui/static/gui/src/app/components/entity-header/entity-header.js

```diff
@@ -76,19 +76,48 @@ YUI.add('entity-header', function() {
     /**
       Callback for failing to get the bundle YAML.
 
-      @method _closeEntityDetails
+      @method _getBundleYAMLFailure
       @param {String} yaml The yaml for the bundle
     */
     _getBundleYAMLFailure: function() {
       //XXX: Need to figure out what to do if there's a failure.
     },
 
+    /**
+      Builds a URL that links to the standalone Juju store, for sharing
+      purposes.
+
+      @method _getStoreURL
+      @param {Object} entity The entity being linked to
+    */
+    _getStoreURL: function(entity) {
+      var url = ['https://jujucharms.com'];
+      if (entity.id.indexOf('~') >= 0) {
+        url.push('u');
+        url.push(entity.owner);
+      }
+      url.push(entity.name);
+      url.push(entity.series);
+      url.push(entity.revision);
+      return encodeURIComponent(url.join('/'));
+    },
+
     render: function() {
       var entity = this.props.entityModel.toEntity();
       var ownerUrl = 'https://launchpad.net/~' + entity.owner;
       var series = entity.series ?
         <li className="entity-header__series">{entity.series}</li> :
         '';
+      var twitterUrl = [
```

Comment:
Yeah, I tried that. The problem is that these are really long strings and they run afoul of our line limit. Rather than putting in an eslint exception I decided to break the strings down into smaller chunks that get joined all together at the end.